backend/src/main/kotlin/app/ehrenamtskarte/backend/cards/webservice/schema/CardMutationService.kt

```diff
@@ -145,9 +145,9 @@ class CardMutationService {
         )
         val userHash = Argon2IdHasher.hashKoblenzUserData(user)
 
-        val userEntitlements = transaction { UserEntitlementsRepository.findUserEntitlements(userHash.toByteArray()) }
+        val userEntitlements = transaction { UserEntitlementsRepository.findByUserHash(userHash.toByteArray()) }
         if (userEntitlements == null || userEntitlements.revoked || userEntitlements.endDate.isBefore(LocalDate.now())) {
```

Comment:
I think we have to throw different exceptions here. At least we should create a TODO and ticket if it will not be done here.
We have to distinguish between the input data does not match the entitlement hash then we throw `UserEntitlementNotFoundException()`

If revoked or expired i would throw
`UserEntitlementExpiredException` or sth similar with a separate message

Reply:
I will need to associate a new exception with some error message.
Do you think `Sie sind nicht mehr berechtigt, einen KoblenzPass zu erstellen` is fine?